contracts/token/ERC721_base.cairo

```diff
@@ -124,6 +129,30 @@ func ERC721_isApprovedForAll{
     return (is_approved)
 end
 
+func ERC721_tokenURI{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(token_id: Uint256) -> (uri_len: felt, uri: felt*):
+    alloc_locals
+    let (exists) = _exists(token_id)
+    assert exists = 1
+
+    let (local base) = ERC721_base_uri.read()
+    let (local uri) = alloc()
+    # without baseURI, should return [0]
+    if base == 0:
+        assert [uri] = 0
+        return (1, uri)
+    end
+    
+    # with baseURI
+    assert [uri] = base
+    assert [uri + 1] = token_id.low
+    assert [uri + 2] = token_id.high
+    return (3, uri)
```

Comment:
I get the idea but this is not actual concatenation (it's filled with zeros and the ASCII representation of a number is not the same number). Also, users will have to deal with three different variables in order to reconstruct the uri. Since this is all too difficult without string operations, @frangio suggested we get rid of the `baseUri` all together and just let users set `tokenUri` directly.

Reply:
Totally valid points. Okay, so we'll set a mapping for tokenID => tokenURI. If I'm understanding this proposal correctly, the tokenURI should be of type felt, correct?—otherwise, we're back to reconstructing with multiple values

Reply:
Right, we should keep it as felt otherwise we're deviating from the standard return value.

Reply:
The standard Solidity return type is string, and IMO this corresponds to felt* because felt has bounded length and string doesn't.

Is there an emerging convention for representing arbitrary-length strings in Cairo?

Reply:
In order to interpret felt* as a string though you need to assume it's null-terminated, i.e. at the first null byte the string should be assumed to end.

Reply:
@frangio I think we should stick to `felt` since we only have short strings by now and I don't think we need to innovate in string representation, but wait for official string support instead

Reply:
Ok! As a data point to make sure this could still be useful: an IPFS uri is around 53 bytes. I believe this would fit fine in a short string.

Reply:
K after discussing it offline with @frangio we agreed the best path forward is to stick with `felt` and then move to another string representation system -- either native support whenever that comes or a library by us or a 3rd party.

We should include this limitation (31 bytes strings) in the documentation since IPFS links wont fit, and we cannot prioritize solving string management to maintain the current focus.